src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/InsertManyCommandResolver.java

```diff
@@ -29,13 +33,28 @@ public Class<InsertManyCommand> getCommandClass() {
 
   @Override
   public Operation resolveCommand(CommandContext ctx, InsertManyCommand command) {
-    final List<WritableShreddedDocument> shreddedDocuments =
-        command.documents().stream().map(doc -> shredder.shred(ctx, doc, null)).toList();
-
-    InsertManyCommand.Options options = command.options();
-    boolean ordered = (null != options) && options.ordered();
-    boolean returnDocumentResponses = (null != options) && options.returnDocumentResponses();
-
-    return new InsertOperation(ctx, shreddedDocuments, ordered, false, returnDocumentResponses);
+    final InsertManyCommand.Options options = command.options();
+    final boolean ordered = (null != options) && options.ordered();
+    final boolean returnDocumentResponses = (null != options) && options.returnDocumentResponses();
+    final List<JsonNode> inputDocs = command.documents();
+    final int docCount = inputDocs.size();
+
+    final List<InsertOperation.InsertAttempt> insertions = new ArrayList<>(docCount);
+    for (int pos = 0; pos < docCount; ++pos) {
+      InsertOperation.InsertAttempt attempt;
+      // Since exception thrown will prevent returning anything, need to instead pass a
+      // reference for Shredder to populate with the document id as soon as it knows it
+      // (there is at least one case fail occurs before it knows the id)
+      AtomicReference<DocumentId> idRef = new AtomicReference<>();
+      try {
+        final WritableShreddedDocument shredded =
+            shredder.shred(ctx, inputDocs.get(pos), null, idRef);
+        attempt = new InsertOperation.InsertAttempt(pos, shredded);
+      } catch (Exception e) {
+        attempt = new InsertOperation.InsertAttempt(pos, idRef.get(), e);
+      }
+      insertions.add(attempt);
+    }
+    return new InsertOperation(ctx, insertions, ordered, false, returnDocumentResponses);
```

Comment:
Can we change this to use `InsertOperation.create()`

Reply:
Yes, just need to add new overload. Done.

Reply:
Ok turns out this was not easy to do, due to type erasure overload (`List` with different element type), so left as-is for now.